### PR TITLE
1.21

### PR DESCRIPTION
**Pull request by Madlykeanu**
update some moved/renamed classes

### CODE REVIEW

File: 1.21/src/main/java/dev/u9g/minecraftdatagenerator/generators/EnchantmentsDataGenerator.java [hunk 1]
```diff
@@ -8,13 +8,18 @@
...
 
 import java.util.List;
+import java.util.Set;
 
 public class EnchantmentsDataGenerator implements IDataGenerator {
-    public static String getEnchantmentTargetName(TagKey<Item> target) {
-        return target.id().getPath().split("/")[1];
+    public static String getEnchantmentTargetName(RegistryEntryList<Item> target) {
+        TagKey<Item> tagKey = target.getTagKey().orElseThrow();
+        return tagKey.toString();
```
Comment (GroobleDierne) [inline 1.21/src/main/java/dev/u9g/minecraftdatagenerator/generators/EnchantmentsDataGenerator.java:22]: Replace with `return tagKey.id().getPath().split("/")[1];` , in this case toString() serializes the class name

File: 1.21/src/main/java/dev/u9g/minecraftdatagenerator/generators/ItemsDataGenerator.java [hunk 2]
```diff
@@ -45,11 +45,12 @@ public static JsonObject generateItem(Registry<Item> itemRegistry, Item item) {
...
 
-        List<TagKey<Item>> enchantmentTargets = getApplicableEnchantmentTargets(item);
+        List<RegistryEntryList<Item>> enchantmentTargets = getApplicableEnchantmentTargets(item);
 
         JsonArray enchantCategoriesArray = new JsonArray();
-        for (TagKey<Item> target : enchantmentTargets) {
-            enchantCategoriesArray.add(EnchantmentsDataGenerator.getEnchantmentTargetName(target));
+        for (RegistryEntryList<Item> target : enchantmentTargets) {
+            String category = EnchantmentsDataGenerator.getEnchantmentTargetName(target);
+            enchantCategoriesArray.add(category);
```
Comment (GroobleDierne) [inline 1.21/src/main/java/dev/u9g/minecraftdatagenerator/generators/ItemsDataGenerator.java:53]: This doesn't work, enchantCategories is always empty.
I have done this code that works but may not be the cleanest, I'm not used to fabric development.
```java
        JsonArray enchantCategoriesArray = new JsonArray();
        Registry<Enchantment> enchants = DGU.getWorld().getRegistryManager().get(RegistryKeys.ENCHANTMENT);
        for (Enchantment enchant : enchants) {
            if (enchant.getApplicableItems().contains(item.getRegistryEntry())) {
                String enchantTarget = enchant.getApplicableItems().getTagKey().get().id().getPath().split("/")[1];
                if (!enchantCategoriesArray.contains(new JsonPrimitive(enchantTarget))) {
                    enchantCategoriesArray.add(enchantTarget);
                }
            }
        }
```

File: 1.21/src/main/java/dev/u9g/minecraftdatagenerator/generators/EnchantmentsDataGenerator.java (hunk 1 above)
Comment (Madlykeanu) [reply 1.21/src/main/java/dev/u9g/minecraftdatagenerator/generators/EnchantmentsDataGenerator.java:22]: ah ok, i made a incorrect assumption based off of the 1.20.5 output that it was supposed to be outputted in the tagkey format

**Comment by Madlykeanu** [comment]
ill update the 1.20.5 enchant category outputs to be correct later

**Review by SuperGamerTron — CHANGES_REQUESTED** [review]
Enchantments are now handled via tags, so we can check whether a given enchantment is in a tag with this method:

```java
private static boolean isEnchantmentInTag(Enchantment enchantment, String tag) {
     return DGU.getWorld()
             .getRegistryManager()
             .get(RegistryKeys.ENCHANTMENT)
             .streamTagsAndEntries()
             .filter(tagKeyNamedPair -> tagKeyNamedPair.getFirst().id().equals(Identifier.of(tag)))
             .flatMap(tagKeyNamedPair -> tagKeyNamedPair.getSecond().stream())
             .anyMatch(enchantmentRegistryEntry -> enchantmentRegistryEntry.value() == enchantment);
}
```

We can use it like this to get the relevant information for each property:
```java
enchantmentDesc.addProperty("treasureOnly", isEnchantmentInTag(enchantment, "treasure"));
enchantmentDesc.addProperty("curse", isEnchantmentInTag(enchantment, "curse"));
enchantmentDesc.addProperty("tradeable", isEnchantmentInTag(enchantment, "tradeable"));
enchantmentDesc.addProperty("discoverable", isEnchantmentInTag(enchantment, "on_random_loot"));
```

**Comment by Madlykeanu** [comment]
Awesome, thank you!

**Pull request merged**